sympy/core/tests/test_numbers.py

```diff
@@ -582,6 +582,14 @@ def test_Float_issue_2107():
     assert S.Zero + b + (-b) == 0
 
 
+def test_issue_14289():
+    from sympy.polys.numberfields import to_number_field
+
+    a = 1 - sqrt(2)
+    b = to_number_field(a)
+    assert b.args[0] == a
```

Comment:
This could be `b.as_expr() == a`. And, as the issue was about the wrong minimal polynomial, that could be tested too by something like `b.minpoly(a).expand() == 0`.